src/renderers/WebGLRenderer.js

```diff
@@ -828,7 +835,8 @@ function WebGLRenderer( parameters = {} ) {
 
 		} else if ( geometry.isInstancedBufferGeometry ) {
 
-			const instanceCount = Math.min( geometry.instanceCount, geometry._maxInstanceCount );
+			const maxInstanceCount = geometry._maxInstanceCount !== undefined ? geometry._maxInstanceCount : Infinity;
+			const instanceCount = Math.min( geometry.instanceCount, maxInstanceCount );
```

Comment:
I don't think we should touch this code block in this PR.

Your example assumes you use `setDrawRange()` with `InstancedBufferGeometry` which should not be necessary. I think we should keep the assumption that `Infinity` means "render everything in the buffer".

Reply:
I can remove the check for `Infinity`, but it won't fix the problem in #19430, since `_maxInstanceCount` will still be `undefined`.

Reply:
I am not sure I understand the use case for an `InstancedBufferGeometry` with `drawCount` set to `Infinity`.

https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/drawArraysInstanced

Even when you are rendering multiple instances, you need to somehow specify what range of vertices you want to render multiple instances of. The correct way of handling #19430 is to specify a vertex count (`drawRange.count`) and a instance count. Maybe there's some use case that I am not seeing...

Reply:
I had to debug this issue more closely to understand what's going on  😅 .

`_maxInstanceCount` is `undefined` in your example since `setupVertexAttributes()` in `WebGLBindingStates` never sets it. That makes sense however since the geometry has no attributes. So you are right, the additional check is actually required.

I've also verified that `Infinity` is not a valid input for all WebGL drawing methods. So it's true that we can't use it as an input.

I'm feeling more confident with the PR now. The only thing that can be changed is the beginning of the change:

```js
const drawRange = geometry.drawRange;
const position = geometry.attributes.position;

let drawStart = drawRange.start * rangeFactor ;
let drawEnd = ( drawRange.start + drawRange.count ) * rangeFactor;
```
We can use this simplification since `drawRange` is always defined when using `BufferGeometry`. It is not a nullable property. So the respective `if` clause can be removed.

Reply:
Yes, this reflects my understanding of things. Initializing `drawStart`/`drawEnd` as you suggest makes sense. I suppose in this instance I was being overly defensive, trying to account for the possibility of `drawRange` being set to `null`. If this will never happen your approach makes more sense, I will adopt it.

Otherwise, I have fairly high confidence that the current code is correct, and that it is a simplification/improvement over the existing code. The only possible alternative that I see for checking `_maxInstanceCount` for `undefined` is to initialize it to `Infinity` in `setupVertexAttributes`. However, I am not sure if that would really be an improvement.